### PR TITLE
migrate to messages api for anthropic, allows for tier checking claude keys

### PR DESCRIPTION
**Pull request by dietpepsigirl**
and an attempt at checking for 'scale' ('custom' rpm/tpm tier corporation keys) as well
matches keys to their appropriate documentation-listed rpm and tpm to match them to a tier, if they don't match, assume it's a custom key. likely not the best implementation but claude keys are hard to come by.

### CODE REVIEW

**Comment by dietpepsigirl** [comment]
As mentioned prev pull request, should likely switch to claude-3-haiku-20240307 to avoid wasting quota on it / also it's faster.

**Comment by kingbased** [comment]
sry github wasn't refreshing dogshit website i didn't see the comments, check my last post

**Comment by dietpepsigirl** [comment]
>i do think msg api integration is good though so the checker doesn't blow up when legacy is deprecated i want to send you some code that you can merge in so it's more elegant how can i contact you

You can contact me on dietpepsi@cocaine.ninja for email (cock.li email address)

**Comment by kingbased** [comment]
never mind what am i doing ive been up for a while im going schizo sorry i don't need to contact for this. can you just move my commit over to the messages api and set the prompt with haiku then ill merge in.

**Comment by dietpepsigirl** [comment]
srry, was eating.
>correct it doesn't, all my keys are like this
Changed your version to messages API, but the 'Pozzed' status is a bit finnicky on Haiku. It was more reliable with text completion instead of messages api, even on claude-2.0 too. I gave it 512 tokens which made it a bit more reliable, but...

**Comment by kingbased** [comment]
sry i was in bed trying to snooze

> So the 25000 tokens limit / remaining is correct in mine, but not in yours.

yeah the text completion api doesn't include the ratelimit headers on any key at all, only the new messages one does. which is why i wanted to bring that change you made over at the very least (you also did work on it and i'd feel bad if i just redid everything). the evaluation stuff is correct though my keys i have are still marked as evaluation even on the new api. just some cleanup things ill point out and then i'll merge it unless you're interested in trying to figure out a prompt for haiku. i'll also take sonnet if you can't get it consistent since it's a little cheaper than claude 2.

File: Anthropic.py [hunk 1]
```diff
@@ -19,48 +8,49 @@ async def check_anthropic(key: APIKey, session):
...
+        ],
+        'temperature': 0.2,
+        'max_tokens': 512
     }
     async with session.post('https://api.anthropic.com/v1/messages', headers=headers, json=data) as response:
         if response.status not in [200, 429, 400]:
             return
 
+        json_response = await response.json()
+
```
Comment (kingbased) [inline Anthropic.py:24]: 3 \n im going to have an aneurysm and start foaming at the mouth

File: Anthropic.py [hunk 2]
```diff
@@ -19,48 +8,49 @@ async def check_anthropic(key: APIKey, session):
...
-            elif error_type == "invalid_request_error":
-                key.has_quota = False #out of quota
+            error_message = json_response.get("error", {}).get("message", "")
+            if "This organization has been disabled" in error_message:
+                return
+            elif "Your credit balance is too low to access the Claude API" in error_message:
+                key.has_quota = False
                 return True
 
+        if key.has_quota:
```
Comment (kingbased) [inline Anthropic.py:38]: don't really need this, has_quota is true by default and the only instance where it's changed to false the function returns early

File: Anthropic.py [hunk 3]
```diff
@@ -74,21 +64,26 @@ def get_tier(tokenlimit, ratelimit):
...
+    for tier, keys_in_tier in keys_by_tier.items():
+        print(f'\n{len(keys_in_tier)} keys found in {tier}:')
+        for key in keys_in_tier:
+            print(f'{key.api_key}' + (' | pozzed' if key.pozzed else "") + (' | rate limited' if key.rate_limited else "") + (' | remaining tokens: ' + str(key.remaining_tokens) if key.remaining_tokens else ""))
 
-    print(f'\n--- Total Valid Anthropic Keys: {len(keys)} ---\n')
+    print(f'\nTotal keys without quota: {len(keys_without_quota)}')
+    for key in keys_without_quota:
+        print(f'{key.api_key}')
+    print(f'\n--- Total Valid Anthropic Keys: {len(keys)} ({len(keys_with_quota)} with quota) ---\n')
```
Comment (kingbased) [inline Anthropic.py:89]: add a \n at the end of file for muh PEP 8 standard slop

**Review by kingbased — COMMENTED** [review]
cool

**Comment by dietpepsigirl** [comment]
Also changed to Sonnet as it's more reliable in my testing and reverted back to 256 tokens due to that
I'll try and figure out a prompt for Haiku but this seems fine for now

**Comment by kingbased** [comment]
ok ty

**Pull request merged**